rust/ip-packet/src/proptests.rs

```diff
@@ -0,0 +1,81 @@
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+use pnet_packet::Packet;
+use proptest::arbitrary::any;
+use proptest::strategy::Strategy;
+
+use crate::make::{tcp_packet, udp_packet};
+use crate::MutableIpPacket;
+
+fn non_icmp_packet() -> impl Strategy<Value = MutableIpPacket<'static>> {
+    (
+        any::<IpAddr>(),
+        any::<IpAddr>(),
+        any::<u16>(),
+        any::<u16>(),
+        any::<Vec<u8>>(),
+        any::<bool>(),
+    )
+        .prop_filter("IPs must be of the same version", |(src,dst,..)| src.is_ipv4() == dst.is_ipv4())
```

Comment:
I reckon make a union instead that either samples a tuple of IPv4 or IPv6 addresses.